### PR TITLE
dev-vcs/datalad: new ebuild

### PR DESCRIPTION
**Pull request by TheChymera**
Builds nicely, and the tests pass if I run `nosetests datalad` a my user after install (except for two, but that's an upstream issue). However, if I run the tests via portage [all are either skipped, or give me errors, or fail](https://bpaste.net/show/67e83f91c78b).

It seems this is at last in part because the tests expect git to be configured (which is teh case for my normal user - but not for the portage user, obviously). @jlec @yarikoptic - how could I make more of these tests work?

### CODE REVIEW

**Comment by yarikoptic** [comment]
You might just like to adopt the way I setup git user in a temp build directory for Debian pkg https://github.com/datalad/datalad/blob/debian/debian/rules

Cheers!

**Comment by TheChymera** [comment]
@yarikoptic ok, that helped, but I still get 13 errors.

See [test output from portage](https://bpaste.net/show/5ffba4d32f80) and [normal test output](https://bpaste.net/show/2c8c1bf76710) - any ideas what else I can tweak?

@jlec will [these lines](https://github.com/TheChymera/neurogentoo/blob/8655f179aaa6d26876e97ca80c7b1097ca3705e9/dev-vcs/datalad/datalad-9999.ebuild#L41-L42) forever modify the git information of the portage user? Or is this somehow rolled back after emerge? If not, what would be the best way (and place) to remove this info?

**Comment by yarikoptic** [comment]
You need to have datalad installed so that entry point scripts were generated, and PATH pointing you them. In Debian package iirc I just use installed for package copy of those and just adjust PATH. Could also be done under virtualenv

**Comment by yarikoptic** [comment]
Ah right, looked back... are that make call and PATH tune up in debian/rules file?

File: dev-vcs/datalad/datalad-9999.ebuild [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	dev-python/keyrings_alt[${PYTHON_USEDEP}]
+	dev-python/progressbar[${PYTHON_USEDEP}]
+	dev-python/requests[${PYTHON_USEDEP}]
+	dev-python/boto[${PYTHON_USEDEP}]
+	dev-python/msgpack[${PYTHON_USEDEP}]
+	dev-vcs/git-annex"
+
+python_test() {
+	git config --global user.name "TESTING"; \
+        git config --global user.email "TESTING@example.com"
```
Comment (jlec) [inline dev-vcs/datalad/datalad-9999.ebuild]: Whitespace error here. PLEASE user repoman and follow the advice.

File: dev-vcs/datalad/datalad-9999.ebuild [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	>=dev-python/keyring-8[${PYTHON_USEDEP}]
+	dev-python/keyrings_alt[${PYTHON_USEDEP}]
+	dev-python/progressbar[${PYTHON_USEDEP}]
+	dev-python/requests[${PYTHON_USEDEP}]
+	dev-python/boto[${PYTHON_USEDEP}]
+	dev-python/msgpack[${PYTHON_USEDEP}]
+	dev-vcs/git-annex"
+
+python_test() {
+	git config --global user.name "TESTING"; \
```
Comment (jlec) [inline dev-vcs/datalad/datalad-9999.ebuild]: no need for ```;\``` and I doubt that ```--global``` really works.

**Comment by TheChymera** [comment]
@yarikoptic This is the tree I get from `distutils_install_for_testing` (the Portage function that is supposed to take care of cases like this):

```
/var/tmp/portage/dev-vcs/datalad-9999/work/datalad-9999-python2_7/
├── bdist.linux-x86_64
├── datalad.egg-info
│   ├── dependency_links.txt
│   ├── entry_points.txt
│   ├── PKG-INFO
│   ├── requires.txt
│   ├── SOURCES.txt
│   └── top_level.txt
├── dist
│   └── datalad-0.2.3-py2.7.egg
├── lib
│   └── datalad
│       ├── api.py
│       ├── auto.py
│       ├── cmdline
│       │   ├── common_args.py
│       │   ├── helpers.py
│       │   ├── __init__.py
│       │   ├── main.py
│       │   └── tests
│       │       ├── __init__.py
│       │       └── test_helpers.py
│       ├── cmd.py
│       ├── config.py
│       ├── consts.py
│       ├── crawler
│       │   ├── base.py
│       │   ├── db.py
│       │   ├── dbs
│       │   │   ├── base.py
│       │   │   ├── files.py
│       │   │   ├── __init__.py
│       │   │   ├── tests
│       │   │   │   ├── __init__.py
│       │   │   │   └── test_files.py
│       │   │   └── versions.py
│       │   ├── __init__.py
│       │   ├── nodes
│       │   │   ├── annex.py
│       │   │   ├── crawl_url.py
│       │   │   ├── __init__.py
│       │   │   ├── matches.py
│       │   │   ├── misc.py
│       │   │   ├── s3.py
│       │   │   └── tests
│       │   │       ├── __init__.py
│       │   │       ├── test_annex.py
│       │   │       ├── test_crawl_url.py
│       │   │       ├── test_matches.py
│       │   │       ├── test_misc.py
│       │   │       └── test_s3.py
│       │   ├── oldconfig
│       │   │   ├── base.py
│       │   │   ├── __init__.py
│       │   │   └── tests
│       │   │       ├── __init__.py
│       │   │       └── test_config.py
│       │   ├── pipeline.py
│       │   ├── pipelines
│       │   │   ├── abstractsonline.py
│       │   │   ├── crcns.py
│       │   │   ├── __init__.py
│       │   │   ├── openfmri.py
│       │   │   ├── openfmri_s3.py
│       │   │   ├── template.py
│       │   │   └── tests
│       │   │       ├── __init__.py
│       │   │       ├── test_openfmri_collection.py
│       │   │       └── test_openfmri.py
│       │   └── tests
│       │       ├── __init__.py
│       │       └── test_pipeline.py
│       ├── customremotes
│       │   ├── archives.py
│       │   ├── base.py
│       │   ├── datalad.py
│       │   ├── __init__.py
│       │   ├── main.py
│       │   └── tests
│       │       ├── __init__.py
│       │       ├── test_archives.py
│       │       └── test_datalad.py
│       ├── db.py
│       ├── distribution
│       │   ├── add_sibling.py
│       │   ├── create_publication_target_sshwebserver.py
│       │   ├── create.py
│       │   ├── dataset.py
│       │   ├── __init__.py
│       │   ├── install.py
│       │   ├── modify_subdataset_urls.py
│       │   ├── move.py
│       │   ├── publish.py
│       │   ├── tests
│       │   │   ├── create_test_dataset.py
│       │   │   ├── __init__.py
│       │   │   ├── test_add_sibling.py
│       │   │   ├── test_dataset_binding.py
│       │   │   ├── test_dataset.py
│       │   │   ├── test_install.py
│       │   │   ├── test_publish.py
│       │   │   ├── test_target_ssh.py
│       │   │   ├── test_uninstall.py
│       │   │   └── test_update.py
│       │   ├── uninstall.py
│       │   └── update.py
│       ├── dochelpers.py
│       ├── downloaders
│       │   ├── base.py
│       │   ├── configs
│       │   │   ├── crawdad.cfg
│       │   │   ├── crcns.cfg
│       │   │   ├── hcp.cfg
│       │   │   ├── openfmri.cfg
│       │   │   └── providers.cfg
│       │   ├── credentials.py
│       │   ├── http.py
│       │   ├── __init__.py
│       │   ├── providers.py
│       │   ├── s3.py
│       │   └── tests
│       │       ├── __init__.py
│       │       ├── test_base.py
│       │       ├── test_credentials.py
│       │       ├── test_http.py
│       │       ├── test_providers.py
│       │       ├── test_s3.py
│       │       └── utils.py
│       ├── __init__.py
│       ├── interface
│       │   ├── add_archive_content.py
│       │   ├── base.py
│       │   ├── clean.py
│       │   ├── crawl.py
│       │   ├── download_url.py
│       │   ├── __init__.py
│       │   ├── ls.py
│       │   ├── save.py
│       │   ├── test.py
│       │   ├── tests
│       │   │   ├── __init__.py
│       │   │   ├── test_add_archive_content.py
│       │   │   ├── test_clean.py
│       │   │   ├── test_crawl.py
│       │   │   ├── test_docs.py
│       │   │   ├── test_download_url.py
│       │   │   ├── test_ls.py
│       │   │   ├── test_save.py
│       │   │   └── test_unlock.py
│       │   └── unlock.py
│       ├── log.py
│       ├── __main__.py
│       ├── resources
│       │   ├── git_ssh.sh
│       │   ├── sshserver_cleanup_after_publish.sh
│       │   └── sshserver_prepare_for_publish.sh
│       ├── support
│       │   ├── annexrepo.py
│       │   ├── archives.py
│       │   ├── cache.py
│       │   ├── configparserinc.py
│       │   ├── constraints.py
│       │   ├── cookies.py
│       │   ├── digests.py
│       │   ├── exceptions.py
│       │   ├── gitrepo.py
│       │   ├── __init__.py
│       │   ├── keyring_.py
│       │   ├── network.py
│       │   ├── param.py
│       │   ├── pprint.py
│       │   ├── protocol.py
│       │   ├── s3.py
│       │   ├── sshconnector.py
│       │   ├── stats.py
│       │   ├── status.py
│       │   ├── strings.py
│       │   ├── tests
│       │   │   ├── __init__.py
│       │   │   ├── test_annexrepo.py
│       │   │   ├── test_cache.py
│       │   │   ├── test_digests.py
│       │   │   ├── test_gitrepo.py
│       │   │   ├── test_sshconnector.py
│       │   │   ├── test_stats.py
│       │   │   ├── test_status.py
│       │   │   ├── test_vcr_.py
│       │   │   └── test_versions.py
│       │   ├── vcr_.py
│       │   └── versions.py
│       ├── tests
│       │   ├── heavyoutput.py
│       │   ├── __init__.py
│       │   ├── test_api.py
│       │   ├── test_archives.py
│       │   ├── test_auto.py
│       │   ├── test_cmdline_main.py
│       │   ├── test_cmd.py
│       │   ├── test_config.py
│       │   ├── test_constraints.py
│       │   ├── test_dochelpers.py
│       │   ├── test_formatters.py
│       │   ├── test_installed.py
│       │   ├── test_interface.py
│       │   ├── test_log.py
│       │   ├── test__main__.py
│       │   ├── test_misc.py
│       │   ├── test_network.py
│       │   ├── test_protocols.py
│       │   ├── test_s3.py
│       │   ├── test_strings.py
│       │   ├── test_testrepos.py
│       │   ├── test_tests_utils.py
│       │   ├── test_utils.py
│       │   ├── test_utils_testrepos.py
│       │   ├── utils.py
│       │   └── utils_testrepos.py
│       ├── ui
│       │   ├── base.py
│       │   ├── dialog.py
│       │   ├── __init__.py
│       │   ├── progressbars.py
│       │   └── tests
│       │       ├── __init__.py
│       │       ├── test_base.py
│       │       └── test_dialog.py
│       ├── utils.py
│       └── version.py
└── test
    └── lib
        ├── datalad-0.2.3-py2.7.egg
        ├── easy-install.pth
        └── site.py
```

What should I add to path?

**Comment by yarikoptic** [comment]
Well,  it installed only the module but not any of the scripts/entry points... as such it is not fully sufficient. May be tomorrow when I get to the laptop I will check the source of that function and nearby definitions if anything could be done to make such install more complete

**Comment by TheChymera** [comment]
@yarikoptic the relevant [dev manual page](https://devmanual.gentoo.org/eclass-reference/distutils-r1.eclass/index.html) says that teh scripts should be installed by `distutils_install_for_testing` under a directory called scripts. However, I cannot find it. @jlec any ideas where the scripts could have gone?

**Comment by yarikoptic** [comment]
looking at https://archives.gentoo.org/gentoo-python/message/c8a8dd9c0e13aab056be4b1b42323b9b since didn't quickly googled up actual source for it I did similar call under a virtualenv to setup.py install directly and it "installed" the scripts:
```shell
> PYTHONPATH=$PWD/venv-gentoostyle/build/lib python setup.py install --home=$PWD/venv-gentoostyle/build --install-lib=$PWD/venv-gentoostyle/build/lib --install-scripts=$PWD/venv-gentoostyle/build/bin
...
$> ls /home/yoh/proj/datalad/datalad/venv-gentoostyle/build/bin
datalad*                   git-annex-remote-datalad-archives*  pbr*
git-annex-remote-datalad*  keyring*                            tqdm*
```
although as you can see also with scripts from other modules which it depends on

**Comment by TheChymera** [comment]
@yarikoptic strange, I really don't see a `scripts` dir appearing in my work tree.

@jlec , @gentoo/python - any ideas why Portage's `distutils_install_for_testing` does not install the scripts for this package?

**Comment by TheChymera** [comment]
@jlec hey again. It seems we can't get portage to run the test suite - but as soon as the software is installed the tests pass and the software works. Can we pull this?

**Comment by yarikoptic** [comment]
worse comes to worse you could just try 'make test' which would install scripts under created `bin/` subdirectory (you might want 'make clean' to clean it away)

**Comment by TheChymera** [comment]
this would best be handled in a new PR given updates in the source code and the knowledge of the submitter.

**Pull request closed without merging**